src/configuration/index.ts

```diff
@@ -45,6 +46,7 @@ export class Configuration implements IConfiguration {
 
 	private static instance: Configuration;
 	private options?: IRequiredConfigOptions;
+	private dot?: any;
```

Comment:
Is dot really from type 'any'? The notation new dot(this.options.dotNotationSeparator) look like a strong notation what it expected.